### PR TITLE
Added convenience initializers for creating new custom datum objects 

### PR DESCRIPTION
**Pull request by SkylerSeamans**
…from custom field objects

### CODE REVIEW

File: TeamSnapSDK/SDK/DataTypes/Leagues/TSDKLeagueCustomDatum.m [hunk 1]
```diff
@@ -23,6 +24,21 @@ + (NSString *)SDKREL {
     return @"league_custom_data";
 }
 
+- (instancetype)initWithField:(TSDKLeagueCustomField *_Nonnull)field memberId:(NSString *_Nonnull)memberId teamId:(NSString *_Nonnull)teamId {
+    self = [super init];
+    if(self) {
+        self.leagueCustomFieldId = [field objectIdentifier];
```
Comment (larsacus) [inline TeamSnapSDK/SDK/DataTypes/Leagues/TSDKLeagueCustomDatum.m]: Do we also need `linkMember`, `linkLeagueCustomField`, and `linkTeam`?

File: TeamSnapSDK/SDK/DataTypes/TSDKCustomDatum.m [hunk 2]
```diff
@@ -13,10 +13,18 @@
...
-// "Check Box"
-// "Text"
-// "Date"
+- (instancetype)initWithField:(TSDKCustomField *_Nonnull)field memberId:(NSString *_Nonnull)memberId teamId:(NSString *_Nonnull)teamId {
+    self = [super init];
+    if(self) {
+        self.memberId = memberId;
+        self.teamId = teamId;
+        self.kind = field.kind;
+        self.options = field.options;
```
Comment (larsacus) [inline TeamSnapSDK/SDK/DataTypes/TSDKCustomDatum.m]: Probably want to shallow copy this array.

File: TeamSnapSDK/SDK/DataTypes/Leagues/TSDKLeagueCustomDatum.m [hunk 3]
```diff
@@ -23,6 +24,21 @@ + (NSString *)SDKREL {
...
+        self.name = field.name;
+        self.options = field.options;
```
Comment (larsacus) [inline TeamSnapSDK/SDK/DataTypes/Leagues/TSDKLeagueCustomDatum.m]: Shallow copy this array.

File: TeamSnapSDK/SDK/DataTypes/Leagues/TSDKLeagueCustomDatum.m (hunk 1 above)
Comment (SkylerSeamans) [reply TeamSnapSDK/SDK/DataTypes/Leagues/TSDKLeagueCustomDatum.m]: I hesitate to do that for a couple reasons:

1) The links sometimes have params specific to the associated object. I'm not sure that we want to associate them.
2) These are convenience initializers for initial Datum creation. We still have to round trip to the server to get actual objects. (see the .app PR that's associated)

**Comment by SkylerSeamans** [comment]
@larsacus addressed

**Pull request merged**